packages/vite/src/node/plugins/html.ts

```diff
@@ -777,6 +784,51 @@ export type IndexHtmlTransform =
       transform: IndexHtmlTransformHook
     }
 
+export function preImportMapHook(
+  config: ResolvedConfig
+): IndexHtmlTransformHook {
+  return (html, ctx) => {
+    const importMapIndex = html.match(importMapRE)?.index
+    if (!importMapIndex) return
+
+    const moduleScriptIndex = html.match(moduleScriptRE)?.index
+    if (!moduleScriptIndex) return
```

Comment:
```suggestion
    const importMapIndex = html.match(importMapRE)?.index
    if (importMapIndex === undefined) return

    const moduleScriptIndex = html.match(moduleScriptRE)?.index
    if (moduleScriptIndex === undefined) return
```

I noticed this edge case. When the html is like below, the warning won't be shown. (`html.match(moduleScriptRE)?.index === 0`)

```html
<script type="module" src="/src/main.js"></script>
<meta charset="UTF-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<title>Vite App</title>

<p>Imported Vue version: <span id="imported-vue-version"></span></p>
<p>Required Vue version: <span id="required-vue-version"></span></p>

<script type="importmap">
  {
    "imports": {
      "vue": "https://unpkg.com/vue@3.2.0/dist/vue.runtime.esm-browser.js"
    }
  }
</script>
```

Other place LGTM 👍